### PR TITLE
Support suggestion for #7854

### LINKED ISSUE #7854: let_underscore_lock does not trigger for `parking_lot` / non-std locks
Lint name: let_underscore_lock

I tried this code:

[playground](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=6de725febdb31b3b08d1bd505006c627)
```rust
fn main() {
    {
        use parking_lot::{Mutex, lock_api::RawMutex}; // 0.11.2
    
        let mutex: Mutex<()> = Mutex::const_new(RawMutex::INIT, ());
        let _ = mutex.lock();
    }
    {
        use std::sync::Mutex;
    
        let mutex: Mutex<()> = Mutex::new(());
        let _ = mutex.lock().unwrap();
    }
}
```

I expected to see this happen: 

Since `parking_lot` is widely used, and usually always the preferred alternative compared to the `std` locks, I expected `clippy` to know about it.

Instead, this happened: 

`clippy` seems to only care about `std` locks:
```
Checking playground v0.0.1 (/playground)
error: non-binding let on a synchronization lock
  --> src/main.rs:13:9
   |
13 |         let _ = mutex.lock().unwrap();
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[deny(clippy::let_underscore_lock)]` on by default
   = help: consider using an underscore-prefixed named binding or dropping explicitly with `std::mem::drop`
   = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#let_underscore_lock

error: could not compile `playground` due to previous error
```

### Meta

```
rustc 1.57.0-nightly (98c861950 2021-09-27)
binary: rustc
commit-hash: 98c8619502093f34ca82f8f26ccf32e753924440
commit-date: 2021-09-27
host: x86_64-pc-windows-msvc
release: 1.57.0-nightly
LLVM version: 13.0.0
```

### PR DESCRIPTION
**Pull request by surechen**
I think the detection of parking_lot's mutex and rwlock is valuable, so submit this pr, please help judge and review, thank you.

Make let_underscore_lock support parking_lot.(Fixes #7854)

changelog: Make let_underscore_lock support parking_lot

### CODE REVIEW

**Review by giraffate — COMMENTED** [review]
Can you add an explanation in the documentation that `parking_lot` is also detected?

FYI we had a little discussion about how to handle external crates:
https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/I'd.20like.20to.20hear.20opinions.20on.20clippy.237957

**Comment by surechen** [comment]
> Can you add an explanation in the documentation that `parking_lot` is also detected?
> 
> FYI we had a little discussion about how to handle external crates: https://rust-lang.zulipchat.com/#narrow/stream/257328-clippy/topic/I'd.20like.20to.20hear.20opinions.20on.20clippy.237957

Done. Thanks for your review.

File: clippy_lints/src/let_underscore.rs [hunk 1]
```diff
@@ -34,6 +34,7 @@ declare_clippy_lint! {
 declare_clippy_lint! {
     /// ### What it does
     /// Checks for `let _ = sync_lock`
+    /// Support mutex and rwlock in std::sync and parking_lot.
```
Comment (giraffate) [inline clippy_lints/src/let_underscore.rs:37]:
```suggestion
    /// Checks for `let _ = sync_lock`.
    /// This supports `mutex` and `rwlock` in `std::sync` and `parking_lot`.
```

**Review by giraffate — COMMENTED** [review]
Sorry for the late review. Overall looks good, I made a small comment.

**Comment by surechen** [comment]
> Sorry for the late review. Overall looks good, I made a small comment.

Done. Thanks.

**Review by giraffate — APPROVED** [review]

**Comment by giraffate** [comment]
@bors r+

Thanks!

**Pull request merged**